plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+			echo __('plugin.importexport.csv.missingHeadersOnCsv', ['missingHeaders' => $missingHeaders]);
+			exit(1);
+		}
+
+		return $file;
+	}
+
+	/**
+	 * @param string $basePath
+	 */
+	private function createAndValidateCSVFileInvalidRows($csvForInvalidRowsName) {
+		$invalidRowsFile = $this->createNewFile($csvForInvalidRowsName, 'w');
+		$invalidRowsFile->fputcsv(array_merge(array_pad($this->_expectedHeaders, $this->_expectedRowSize, null), ['error']));
+
+		return $invalidRowsFile;
+	}
+
+	/**
+	 * Insert static data that will be used for the submission processing
+	 */
+	private function initializeStaticVariables() {
+		$this->_dirNames = Application::getFileDirectories();
+		$this->_format = trim($this->_dirNames['context'], '/') . '/%d/' . trim($this->_dirNames['submission'], '/') . '/%d';
+
+		$this->_fileManager = new FileManager();
+		$this->_publicFileManager = new PublicFileManager();
+
+		$this->_fileService = Services::get('file');
+		$this->_publicationService = Services::get('publication');
+	}
+
+	/**
+	 * Returns a cached DAO or create a new one, if it isn't initialized yet.
+	 *
+	 * @param string $daoType
+	 */
+	private function getCachedDao($daoType) {
+		return $this->_daos[$daoType] ?? $this->_daos[$daoType] = DAORegistry::getDAO($daoType);
+	}
+
+	/**
+	 * Returns a cached Press or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedPress($pressPath) {
+		return $this->_presses[$pressPath] ?? $this->_presses[$pressPath] = $this->getPressByPath($pressPath);
+	}
+
+	/**
+	 * Returns a cached Genre or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedGenreId($pressId) {
+		return $this->_genreIds[$pressId] ?? $this->_genreIds[$pressId] = $this->getGenreIdByPressId($pressId);
+	}
+
+	/**
+	 * Returns a cached Serie ID or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $seriesPath
+	 * @param int $pressId
+	 */
+	private function getCachedSerieId($seriesPath, $pressId) {
+		$key = "{$seriesPath}_{$pressId}";
+		return $this->_serieIds[$key] ?? $this->_serieIds[$key] = $this->getSerieIdByPathAndPressId($seriesPath, $pressId);
+	}
+
+	/**
+	 * Returns a cached UserGroup ID or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedUserGroupId($pressId) {
+		return $this->_userGroupIds[$pressId] ?? $this->_userGroupIds[$pressId] = $this->getUserGroupIdByPressId($pressId);
+	}
+
+	/**
+	 * Insert data on the invalid_rows.csv file and increase the failed rows counter
+	 *
+	 * @param SplFileObject $invalidRowsFile
+	 * @param array $fields
+	 * @param int $failedRows
+	 * @param string $reason
+	 */
+	private function processFailedRow($invalidRowsFile, $fields, $reason, $failedRows) {
+		$invalidRowsFile->fputcsv(array_merge(array_pad($fields, $this->_expectedRowSize, null), [$reason]));
+
+		return $failedRows + 1;
```

Comment:
If you receive the `$failedRows` by reference, then you can replace the occurrences of `$failedRows = $this->processFailedRow(...)` by `$this->processFailedRow(...)`

```suggestion
	private function processFailedRow($invalidRowsFile, $fields, $reason, &$failedRows) {
		$invalidRowsFile->fputcsv(array_merge(array_pad($fields, $this->_expectedRowSize, null), [$reason]));
		++$failedRows;
```

Reply:
But as you've created a couple of private variables, you could also create one for the `$failedRows` and the `$invalidRowsFile`, then you won't need to pass them by arguments.